### PR TITLE
[RW-9090][risk=no] Add handles to UI to improve test reliability

### PR DESCRIPTION
**Pull request by dmohs**
Add `aria-label` and `id` attributes to UI elements as handles for CSS selectors in tests.

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: e2ev2/src/config.js [hunk 1]
```diff
@@ -10,10 +10,10 @@ export_({usernames})
 
 const urlRoot = () => {
   assert(
-    process.env.UI_HOSTNAME,
-    'UI_HOSTNAME not defined. Try: export UI_HOSTNAME=all-of-us-workbench-test.appspot.com'
+    process.env.UI_URL_ROOT,
+    'UI_URL_ROOT not defined. Try: export UI_URL_ROOT=https://all-of-us-workbench-test.appspot.com'
   )
-  return `https://${process.env.UI_HOSTNAME}`
+  return process.env.UI_URL_ROOT
```
Comment (dmohs) [inline e2ev2/src/config.js:16]: Unrelated change to make it easier to run tests against, e.g., http://localhost:4200.

File: e2ev2/tests/user-create.test.js [hunk 2]
```diff
@@ -15,34 +12,36 @@ const pressKey = async (page, keyName, count = 1) => {
...
+  await tu.jsClick(page, ackCheckboxSel)
+  await page.click('[role="button"][aria-label="Next"]')
 
-  await page.waitForSelector('.p-dropdown') // Institution Page
-  await page.click('.p-dropdown')
-  await page.waitForSelector('[role="option"][aria-label="Broad Institute"]')
-  await page.click('[role="option"][aria-label="Broad Institute"]')
+  await page.waitForSelector('#account-creation-institution') // Institution Page
+  await page.waitForSelector('[aria-label="Institution"]')
+    .then(eh => eh.evaluate(e => e.parentNode.parentNode.click()))
```
Comment (dmohs) [inline e2ev2/tests/user-create.test.js:33]: This one is ugly because we have limited control over PrimeReact components.

File: e2ev2/tests/user-create.test.js [hunk 3]
```diff
@@ -15,34 +12,36 @@ const pressKey = async (page, keyName, count = 1) => {
...
+  await page.waitForSelector('[role="option"][aria-label="Broad Institute"]').then(eh => eh.click())
```
Comment (dmohs) [inline e2ev2/tests/user-create.test.js:34]: Improvement suggested by Joel on previous PR.

File: e2ev2/src/config.js (hunk 1 above)
Comment (dmohs) [inline e2ev2/src/config.js:16]: Unrelated fix to make it easier to run against, e.g., http://localhost:4200.

File: ui/src/app/components/terms-of-service.tsx [hunk 4]
```diff
@@ -132,6 +132,7 @@ export class TermsOfService extends React.Component<TosProps, TosState> {
             </div>
             <div>
               <CheckBox
+                aria-label='Acknowledge Terms'
                 data-test-id='agreement-check'
```
Comment (NehaBroad) [inline ui/src/app/components/terms-of-service.tsx:136]: do we still need data-test-id if its being replaced in test with input aria-label

File: e2ev2/tests/workspace-create.test.js [hunk 5]
```diff
@@ -33,40 +33,36 @@ browserTest('create a workspace', async browser => {
...
   const workspacePageTitleLink =
-    await page.waitForSelector('a[href="/workspaces"] + span + div > a',
-    {timeout: workspaceCreationTimeoutMs})
+    await page.waitForSelector('a[href="/workspaces"] + span + div > a')
   await expect(workspacePageTitleLink.evaluate(e => e.innerText)).resolves.toBe(wsName)
-  await page.click('[data-test-id="workspace-menu-button"]')
+  await page.click('[aria-label="Open Actions Menu"]')
   // side menu pops up
-  await page.click('div#popup-root clr-icon[shape="trash"]')
+  await page.click('[aria-label="Delete"]')
```
Comment (NehaBroad) [inline e2ev2/tests/workspace-create.test.js:56]: Oh wow! Thats clean and so easy to understand what the action is for

File: e2ev2/tests/workspace-create.test.js [hunk 6]
```diff
@@ -2,7 +2,7 @@ const fsp = require('fs').promises
 const config = require('../src/config')
 const impersonate = require('../src/impersonate')
 const tu = require('../src/test-utils')
-const utils = require('../src/utils')
+const u = require('../src/utils')
```
Comment (NehaBroad) [inline e2ev2/tests/workspace-create.test.js:5]: Why u and not utils?

**Review by NehaBroad — APPROVED** [review]
This is really nice! The tests are so easy to understand. My only concern is that we should note somewhere the intent of using id and not data-test-id, so that the dev can be encouraged to make the right decision for new code.

File: ui/src/app/components/terms-of-service.tsx (hunk 4 above)
Comment (dmohs) [reply ui/src/app/components/terms-of-service.tsx:136]: We should not, but I didn't want to risk taking them out just yet. I'll see about doing another PR to take out any of these that I can to help establish the new pattern.
